### DISCUSSION TITLE
Explicit nuget package reference required for derived class

### DISCUSSION THREAD

**Discussion by koluradev** [ControlzEx/ControlzEx#188] [Q&A] [answered]
I am using ControlzEx nuget package in a project A.csproj and have added a derived class 
```
public class MyWindow : WindowChromeWindow 
{
    public string MyPropertry { get; set; }
}
```

Now I want to use this derived window in another assembly B.exe as the main window. 
The project B.csproj has an assembly reference to A.dll (not a project reference).

When I try to build project B I get an error saying
```
error MC1000: Unknown build error, 'Could not find assembly 'ControlzEx, Version=6.0.0.0, Culture=neutral, PublicKeyToken=69f1c32f803d307e'. 
Either explicitly load this assembly using a method such as LoadFromAssemblyPath() or use a MetadataAssemblyResolver that returns a valid assembly.'
```
allthough ControlzEx.dll exists in output folder next to B.exe. 
Furthermore, Visual Studio can only resolve `MyProperty` but no base properties from `Window` class like `Title`. If I derive directly from `Window` those base properties are resolved. 

The error will only disappear and properties are resolved if I add the ControlzEx nuget package as a direct reference to B.csproj.

Since I've done this kind of enhancing external controls many times without problems, I don't understand why the additional nuget reference in B.csproj is required.
It would be great if someone had an idea or solution for this.

**Comment by batzen** [comment]
Could you provide a repro project?
There is nothing special in ControlzEx that could cause such issues, but maybe i can find out what's causing your issue when having a repro.

**Comment by koluradev** [comment]
Sorry for late response, returning to this project after two weeks. 
I set up a simple repository where the problem is reproducible and added you to this repo.

**Comment by batzen** [comment, marked as answer]
@redstripes89 I could reproduce your issue using your repro.

After some pocking i added `<RestoreProjectStyle>PackageReference</RestoreProjectStyle>` to both projects, which resolved the MC1000 error. So this seems kind of related to the "old" csproj format.
Afterwards i still got compile errors because the namespace for your "App" project was "App" and the main application class was also named "App" this confused the compiler and it complained about being unable to find the class "App.App".
After changing the namespace to "MyApp" and replacing it's usages everything compiles fine.

So it's not an ControlzEx issue. ;-)

**Comment by koluradev** [comment]
Thank you for clarification. Will check this with our project and see if the new csproj format can avoid this issue.
